vista_2d/components.py

```diff
@@ -0,0 +1,352 @@
+            else:
+                outdir = self.output_dir
+
+            outname = os.path.join(outdir, basename + ".tif")
+
+            label = seg.cpu().numpy()
+            lm = label.max()
+            if lm <= 255:
+                label = label.astype(np.uint8)
+            elif lm <= 65535:
+                label = label.astype(np.uint16)
+            else:
+                label = label.astype(np.uint32)
+
+            tifffile.imwrite(outname, label)
+
+            print(
+                f"Saving {outname} shape {label.shape} max {label.max()} dtype {label.dtype}"
+            )
+
+        return d
+
+
+class LabelsToFlows(MapTransform):
+    # based on dynamics labels_to_flows()
+    # created a 3 channel output (foreground, flowx, flowy) and saves under flow (new) key
+
+    def __init__(self, flow_key, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.flow_key = flow_key
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.key_iterator(d):
+            label = d[key].int().numpy()
+
+            label = fastremap.renumber(label, in_place=True)[0]
+            veci = masks_to_flows(label[0], device=None)
+
+            flows = np.concatenate((label > 0.5, veci), axis=0).astype(np.float32)
+            flows = convert_to_dst_type(
+                flows, d[key], dtype=torch.float, device=d[key].device
+            )[0]
+            d[self.flow_key] = flows
+            # meta_data = {ImageMetaKey.FILENAME_OR_OBJ : filename}
+            # d[key] = MetaTensor.ensure_torch_and_prune_meta(img_array, meta_data)
+        return d
+
+
+class LogitsToLabels:
+    def __call__(self, logits, filename=None):
+        device = logits.device
+        logits = logits.float().cpu().numpy()
+        dP = logits[1:]  # vectors
+        cellprob = logits[0]  # foreground prob (logit)
+
+        try:
+            pred_mask, p = compute_masks(
+                dP,
+                cellprob,
+                niter=200,
+                cellprob_threshold=0.4,
+                flow_threshold=0.4,
+                interp=True,
+                device=device,
+            )
+        except RuntimeError as e:
+            logger.warning(
+                f"compute_masks failed on GPU retrying on CPU {logits.shape} file {filename} {e}"
+            )
+            pred_mask, p = compute_masks(
+                dP,
+                cellprob,
+                niter=200,
+                cellprob_threshold=0.4,
+                flow_threshold=0.4,
+                interp=True,
+                device=None,
+            )
+
+        return pred_mask, p
+
+
+class LogitsToLabelsd(MapTransform):
+    def __call__(self, data):
+        d = dict(data)
+        f = LogitsToLabels()
+        for key in self.key_iterator(d):
+            pred_mask, p = f(d[key])
+            d[key] = pred_mask
+            d[f"{key}_centroids"] = p
+        return d
+
+
+class SaveTiffExd(MapTransform):
```

Comment:
I didn't find where use this class.

Reply:
It is not being used in the tutorial notebook, do you want me to remove that? I feel it can be useful if the user wants to save an image etc. @KumoLiu